specs/directory.md

```diff
@@ -0,0 +1,395 @@
+# Notation Directory Structure
+
+The `notation` CLI requires local file systems support for the following components across various platforms.
+
+- `notation` binary
+- Plugins
+- Configurations
+- Trust stores
+- Trust policies
+- Signature caches
+- Signing key store
+
+This documentation specifies the recommended directory structure for those components.
+
+## Category
+
+The directories for various components are classified into the following catagories.
+
+| Alias     | Description                                                                         |
+| --------- | ----------------------------------------------------------------------------------- |
+| `BIN`     | Directory for executable binaries                                                   |
+| `LIBEXEC` | Directory for binaries not meant to be executed directly by users' shell or scripts |
+| `CONFIG`  | Directory for configurations                                                        |
+| `CACHE`   | Directory for user-specific cache                                                   |
+
+Although it is recommended to install `notation` with its plugins and default configurations at the system level, it is possible to install at the user level.
+
+On Unix systems, `notation` follows [Filesystem Hierarchy Standard][FHS] for system level directories and [XDG Base Directory Specification][XDG] for user level directories. On Windows, [Known Folders][KF] and [App Settings][AS] are followed equivalently. On Darwin, [macOS File System][macOS_FS] with [System Integrity Protection][SIP] is followed equivalently. If a file with the same name exists at the system level and the user level, the file at the user level takes over the priority.

```

Comment:
This makes sense from a usability perspective for certain components like config.json, but seems inverse of what is desired for trust store and trust policy.

Reply:
```suggestion
On Unix systems, `notation` follows [Filesystem Hierarchy Standard][FHS] for system level directories and [XDG Base Directory Specification][XDG] for user level directories. On Windows, [Known Folders][KF] and [App Settings][AS] are followed equivalently. On Darwin, [macOS File System][macOS_FS] with [System Integrity Protection][SIP] is followed equivalently.

**TODO** : If a file with the same name exists at the system level and the user level, the mechanism to resolve priority needs to be defined (tracked in #203).

```